Source/WebCore/editing/Editor.cpp

```diff
@@ -2105,8 +2105,40 @@ void Editor::selectComposition()
     document().selection().setSelection(selection, { });
 }
 
+Element* Editor::selectedElement()
+{
+    Ref document = protectedDocument();
+
+    RefPtr node = document->selection().selection().end().protectedDeprecatedNode();
```

Comment:
Is the intent of this code to find the closest ancestor `Element` that contains the current selection? If so, we may want to consider using `containerNode()`, which handles the `BeforeAnchor` and `AfterAnchor` cases correctly.

If this is really about the anchor node, then `protectedAnchorNode()` or `anchorNode()` should suffice.